go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -519,74 +505,104 @@ func getPolicyType(policy *knet.NetworkPolicy) (bool, bool) {
 
 // getNewLocalPolicyPorts will find and return port info for every given pod obj, that is not found in
 // np.localPods.
-// if there are problems with fetching port info from logicalPortCache, pod will be added to errObjs.
-func (oc *DefaultNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
-	objs ...interface{}) (policyPortsToUUIDs map[string]string, policyPortUUIDs []string, errObjs []interface{}) {
+// if there are problems with fetching port info from logicalPortCache, error will be added to returned error array.
+func (bnc *BaseNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
+	objs ...interface{}) (map[string]string, []string, []error) {
+	var errs []error
 
-	policyPortUUIDs = make([]string, 0, len(objs))
-	policyPortsToUUIDs = map[string]string{}
+	policyPortUUIDs := []string{}
+	policyPortsToUUIDs := map[string]string{}
 
 	for _, obj := range objs {
 		pod := obj.(*kapi.Pod)
-
-		logicalPortName := util.GetLogicalPortName(pod.Namespace, pod.Name)
-		if _, ok := np.localPods.Load(logicalPortName); ok {
-			// port is already added for this policy
-			continue
-		}
-
 		if pod.Spec.NodeName == "" {
 			// pod is not yet scheduled, will receive update event for it
 			continue
 		}
 
 		// Skip pods that will never be present in logicalPortCache,
 		// e.g. hostNetwork pods, overlay node pods, or completed pods
-		if !oc.podExpectedInLogicalCache(pod) {
+		if !bnc.podExpectedInLogicalCache(pod) {
 			continue
 		}
 
-		// Add pod to errObjs for retry if
-		// 1. getting pod LSP from the cache fails,
-		// 2. the gotten LSP is scheduled for removal (stateful-sets).
-		portInfo, err := oc.logicalPortCache.get(pod, types.DefaultNetworkName)
-		if err != nil {
-			klog.Warningf("Failed to get LSP for pod %s/%s for networkPolicy %s, err: %v",
-				pod.Namespace, pod.Name, np.name, err)
-			errObjs = append(errObjs, pod)
-			continue
+		nadNames := bnc.getPodNADNames(pod)
+		for _, nadName := range nadNames {
+			portInfo, err := bnc.getNewLocalPolicyPortsForNAD(np, pod, nadName)
```

Comment:
I guess at this point it makes sense to inline `getNewLocalPolicyPortsForNAD` back to this function, since it is not called from anywhere else?